File: client/src/types/assessment.ts
import { JsonValue } from "./renderers";

export interface Assessment {
  assessment_id?: string;
  name: string;
  value: JsonValue;
  type?: 'feedback' | 'expectation';
  rationale?: string;
  metadata?: Record<string, JsonValue>;
  source?: {
    source_type: string;
    source_id: string;
  } | string;
  timestamp?: string;
}

File: server/models/assessment.py
from typing import Any, Dict, Optional

from pydantic import BaseModel


class AssessmentSource(BaseModel):
  """Source of an assessment."""

  source_type: str  # 'HUMAN', 'LLM_JUDGE', etc.
  source_id: str  # username or model id


class Assessment(BaseModel):
  """MLflow trace assessment with value and rationale."""

  assessment_id: Optional[str] = None
  name: str
  value: Any
  type: Optional[str] = None  # 'feedback' or 'expectation'
  rationale: Optional[str] = None
  metadata: Optional[Dict[str, Any]] = None
  source: Optional[AssessmentSource] = None
